src/app/components/InlineLink/README.md

```diff
@@ -0,0 +1,55 @@
+# InlineLink
+
+## Description
+
+The `InlineLink` component is used to render link elements containting text. It specifically does not render content using the `children` prop and instead uses a prop called `text` so that only a string of text can be passed in. This is to ensure that, for any external links, an `aria-label` can be rendered containing the text content with a localised "external" message to indicate to assistive technology users that the link they are reading will navigate away from a BBC domain.
+
+If you need a link component to wrap content that is not just a string of text, you should use the [`Link`](../Link/README.md) component. The `InlineLink` component extends the [`Link`](../Link/README.md) component which is responsible for determining whether to make a client-side or server-side request.
```

Comment:
Just looking at the `Link` component, is there anyway we could disallow its use with just a string a child with types? Just wary people will accidently use `Link` when they should be using `InlineLink`? It's also not completely clear on `Link` that it shouldn't be used for text and this component should be used instead.

Reply:
@andrewscfc I tried using types to disallow children that is not a string but the issue is children is always a type of React.Element so this was not possible.

With regards to the Link component, I thought it was best to just if Link didn't exist so I put the Link logic in InlineLink and deleted Link.